.github/workflows/python-publish.yml

```diff
@@ -17,12 +17,12 @@ jobs:
         python-version: '3.x'
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
+        python -m pip install --upgrade pip build
         pip install setuptools twine
     - name: Build and publish
       env:
         TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
         TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
       run: |
-        python setup.py sdist
+        python -m build
```

Comment:
I think you are missing a `pip`

Reply:
[This is the current recommended way of building packages](https://packaging.python.org/tutorials/packaging-projects/#generating-distribution-archives)